src/winacl/winacl.c

```diff
@@ -333,7 +354,30 @@ merge_acl(acl_t acl, acl_t *prev_acl, const char *path)
 
 	return (0);
 }
-
+ 
+/* add inherited flag to ACES in ACL */
+static int
+set_inherited_flag(acl_t *acl)
+{
+        int entry_id;
+        acl_entry_t acl_entry;
+        acl_flagset_t acl_flags;
+                 
+        entry_id = ACL_FIRST_ENTRY;
+        while (acl_get_entry(*acl, entry_id, &acl_entry) > 0) {
+                entry_id = ACL_NEXT_ENTRY;
+
+                if (acl_get_flagset_np(acl_entry, &acl_flags) < 0)
+                        err(EX_OSERR, "acl_get_flagset_np() failed");
+ 
+                acl_add_flag_np(acl_flags, ACL_ENTRY_INHERITED);
```

Comment:
Not an ACL expert but shouldnt we verify if that flag is not set already to avoid calling the set function if nothing changes? Or does acl_set_flagset_np avoid hitting the disk in such cases?